core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2484,6 +2497,259 @@ private long startOffsetDuringInitialization(long partitionDataStartOffset) thro
         }
     }
 
+    private ShareAcquiredRecords filterAbortedTransactionalAcquiredRecords(FetchPartitionData fetchPartitionData, FetchIsolation isolationLevel, ShareAcquiredRecords shareAcquiredRecords) {
+        if (isolationLevel != FetchIsolation.TXN_COMMITTED)
+            return shareAcquiredRecords;
+        // When FetchIsolation.TXN_COMMITTED is used as isolation type by the share group, we need to filter any
```

Comment:
Why not to have the below check in the same method and pass the aborted transactions to `filterAbortedTransactionalRecords` without optional. This can make sure that when it's required to filter then only call goes to further methods, and maybe method should have all these pre-checks.

```
if (abortedTransactions.isEmpty())
                return acquiredRecords;